### PR TITLE
add box height and width responsive sizes

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

Box `height` and `width` accept an array of valid sizes to support a responsive layout. 

## More Information

Fixes: #16110 

### Before & After Video

https://www.loom.com/share/dd41599bc3cc4b99baf4e7bab7c2d45f

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [x] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

### CODE REVIEW

**Review by georgewrmarshall — COMMENTED** [review]
Looks good! So stoked this is being fixed. Do we need to add any more unit tests?

**Review by georgewrmarshall — DISMISSED** [review]
LGTM

File: ui/components/ui/box/box.js [hunk 1]
```diff
@@ -23,20 +23,29 @@ const BASE_CLASS_NAME = 'box';
...
+const ValidBlockSize = PropTypes.oneOf(BLOCK_SIZES);
 const ValidSizeAndAuto = PropTypes.oneOf([...Sizes, 'auto']);
 export const ValidBackgroundColors = PropTypes.oneOf(
   Object.values(BACKGROUND_COLORS),
 );
 export const ValidBorderColors = PropTypes.oneOf(Object.values(BORDER_COLORS));
 export const ValidTextColors = PropTypes.oneOf(Object.values(TEXT_COLORS));
 export const ValidIconColors = PropTypes.oneOf(Object.values(ICON_COLORS));
+const ValidAlignItem = PropTypes.oneOf(ALIGN_ITEMS);
+const ValidJustifyContent = PropTypes.oneOf(JUSTIFY_CONTENT);
```
Comment (brad-decker) [inline ui/components/ui/box/box.js:35]: the Object.values() was dropped here which I think should be in place still?

**Review by brad-decker — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Review by georgewrmarshall — COMMENTED** [review]
Would it be beneficial to get some stories showing some layout options using the shorthand prop notation for widths?

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**